src/D2L.Bmx/ParentProcess.cs

```diff
@@ -2,43 +2,41 @@
 using System.Runtime.InteropServices;
 
 namespace D2L.Bmx;
+
 /**
 This file may eventually be replaced if the following dotnet runtime api proposals are implemented:
 - https://github.com/dotnet/runtime/issues/21941
 - https://github.com/dotnet/runtime/issues/24423
-
-Uses the same approach of calling NtQueryInformationProcess as in the PowerShell library
-https://github.com/PowerShell/PowerShell/blob/26f621952910e33840efb0c539fbef1e2a467a0d/src/System.Management.Automation/engine/ProcessCodeMethods.cs
 */
-internal partial class WindowsParentProcess {
+internal partial class ParentProcess {
+	public static string? GetParentProcessName() {
+		int parentPid = -1;
+		if( OperatingSystem.IsWindows() ) {
+			parentPid = GetWindowsParentProcessId();
+		} else if( OperatingSystem.IsLinux() || OperatingSystem.IsMacOS() ) {
+			parentPid = GetPosixParentProcessId();
+		}
+		if( parentPid == -1 ) {
+			return null;
+		}
+		return Process.GetProcessById( parentPid ).ProcessName;
+	}
 
-	[LibraryImport( "ntdll.dll", EntryPoint = "NtQueryInformationProcess" )]
-	internal static partial int NtQueryInformationProcess(
```

Comment:
Moved below, as per our convention to put public/high-level methods first, and private/low-level/helper methods after.
Also changed it to `private`, as it's not used elsewhere.